src/main/java/appeng/core/sync/packets/PacketCompressedNBT.java

```diff
@@ -28,93 +28,108 @@
 import java.util.zip.GZIPOutputStream;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiScreen;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.CompressedStreamTools;
 import net.minecraft.nbt.NBTTagCompound;
 
-import cpw.mods.fml.relauncher.Side;
-import cpw.mods.fml.relauncher.SideOnly;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.client.gui.implementations.GuiInterfaceTerminal;
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
+import appeng.helpers.Reflected;
 
 
-public class PacketCompressedNBT extends AppEngPacket
+public class PacketCompressedNBT implements AppEngPacket, AppEngPacketHandler<PacketCompressedNBT, AppEngPacket>
 {
 
 	// input.
-	final NBTTagCompound in;
+	private NBTTagCompound in;
 	// output...
-	private final ByteBuf data;
-	private final GZIPOutputStream compressFrame;
-	int writtenBytes = 0;
-	boolean empty = true;
+	private ByteBuf data;
+	private GZIPOutputStream compressFrame;
 
-	// automatic.
-	public PacketCompressedNBT( final ByteBuf stream ) throws IOException
+	@Reflected
+	public PacketCompressedNBT()
+	{
+		// automatic.
+	}
+
+	// api
+	public PacketCompressedNBT( final NBTTagCompound din )
+	{
+		this.in = din;
+	}
+
+	@Override
+	public void fromBytes( final ByteBuf buf )
 	{
 		this.data = null;
 		this.compressFrame = null;
 
-		final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
+		final GZIPInputStream gzReader;
+		try
 		{
-
-			@Override
-			public int read() throws IOException
+			gzReader = new GZIPInputStream( new InputStream()
 			{
-				if( stream.readableBytes() <= 0 )
+
+				@Override
+				public int read() throws IOException
 				{
-					return -1;
+					if( buf.readableBytes() <= 0 )
+					{
+						return -1;
+					}
+
+					return buf.readByte() & 0xff;
 				}
+			} );
 
-				return stream.readByte() & 0xff;
-			}
-		} );
+			final DataInputStream inStream = new DataInputStream( gzReader );
+			this.in = CompressedStreamTools.read( inStream );
+			inStream.close();
+		}
+		catch( final IOException e )
+		{
+			// TODO Auto-generated catch block
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)